sdk/src/pubkey.rs

```diff
@@ -99,8 +103,34 @@ impl Pubkey {
             hasher.hash(seed);
         }
         hasher.hashv(&[program_id.as_ref(), "ProgramDerivedAddress".as_ref()]);
+        let hash = hash(hasher.result().as_ref());
 
-        Ok(Pubkey::new(hash(hasher.result().as_ref()).as_ref()))
+        if curve25519_dalek::edwards::CompressedEdwardsY::from_slice(hash.as_ref())
+            .decompress()
+            .is_some()
+        {
+            return Err(PubkeyError::InvalidSeeds);
+        }
+
+        Ok(Pubkey::new(hash.as_ref()))
+    }
+
+    /// Find a valid program address and its corresponding nonce which must be passed
+    /// as an additional seed when calling `create_program_address`
+    #[cfg(not(feature = "program"))]
+    pub fn find_program_address(seeds: &[&[u8]], program_id: &Pubkey) -> (Pubkey, u8) {
```

Comment:
@jackcmay are you betting that u8 is a big enough nonce?

Reply:
@aeyakovenko Yes